File: pa3/DictionaryTest.java
/*------------------------------------------------------------------------------
Ly Phung 
lynphung
12B
4/27/19
DictionaryTest.java that tests the Dictionary.java file
DictionaryTest.java
------------------------------------------------------------------------------*/

class DictionaryTest
{
	public static void main(String[] args)
	{
		Dictionary dict = new Dictionary();

		//test methods
		dict.insert("1", "69");
		System.out.println(dict.toString());
		dict.insert("2", "420");
		System.out.println(dict.toString());
		dict.insert("3", "yes boiiiiii");
		
		//lookup 1st term
		System.out.println(dict.lookup("1"));
		//lookup middle term
		System.out.println(dict.lookup("2"));
		//lookup last term
		System.out.println(dict.lookup("3"));
		System.out.println();
		System.out.println(dict.toString());
		
		//delete first item
		dict.delete("1");
		//middle item
		//dict.delete("2");
		//last item
		//dict.delete("3");
		System.out.println(dict.toString());
		
		dict.insert("4", "dictionary");
		System.out.println(dict.toString());
		
		dict.makeEmpty();
		System.out.println(dict.toString());
		
		//test keynotfound
		//dict.delete("7");
		
		//test duplicatekey
		//dict.insert("4", "notdictionary");
	}
}

File: lab5/DictionaryTest.c
/*
* DictionaryTest.c
* Ly Phung
* lynphung
* 12M
* 5/11/19
* Test file for Dictionary.h
*/

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<assert.h>
#include"Dictionary.h"

int main(int argc, char* argv[])
{
	Dictionary A = newDictionary();
   	
   	printf("%s\n", (isEmpty(A)?"true":"false"));
   	printf("%d\n", size(A));

   	insert(A, "1", "poop");
   	printDictionary(stdout, A);

   	insert(A, "2", "yes");
   	printDictionary(stdout, A);
   	insert(A, "3", "hahahhahhhhahaha");
   	printf("%s\n", (isEmpty(A)?"true":"false"));
   	printf("%d\n", size(A));

   	//printf("key=\"%s\" %s\"%s\"\n", "1", (lookup(A, "1") ==NULL?"not found ":"value="), lookup(A, "1"));
   	//printf("key=\"%s\" %s\"%s\"\n", "2", (lookup(A, "2") ==NULL?"not found ":"value="), lookup(A, "2"));
   	//printf("key=\"%s\" %s\"%s\"\n", "3", (lookup(A, "3") ==NULL?"not found ":"value="), lookup(A, "3"));
   	printDictionary(stdout, A);

   	delete(A, "1");
   	//delete(A, "2");
   	//delete(A, "3");
   	printDictionary(stdout, A);
   	printf("%s\n", (isEmpty(A)?"true":"false"));
   	printf("%d\n", size(A));

   	makeEmpty(A);
   	printf("%s\n", (isEmpty(A)?"true":"false"));
   	printf("%d\n", size(A));

   	freeDictionary(&A);

   	return(EXIT_SUCCESS);
}